pallets/messages/src/lib.rs

```diff
@@ -344,28 +351,39 @@ impl<T: Config> Pallet<T> {
 		schema_id: SchemaId,
 		current_block: BlockNumberFor<T>,
 	) -> Result<bool, DispatchError> {
-		<Messages<T>>::try_mutate(
-			current_block,
-			schema_id,
-			|existing_messages| -> Result<bool, DispatchError> {
-				// first message for any schema_id is going to trigger an event
-				let need_event = existing_messages.len() == 0;
-				let index = BlockMessageIndex::<T>::get();
-				let msg = Message {
-					payload, // size is checked on top of extrinsic
-					provider_msa_id,
-					msa_id,
-					index,
-				};
+		let mut metadata = MessageBlockMetadata::<T>::get();
+		ensure!(
+			u32::from(metadata.total_index) < T::MaxMessagesPerBlock::get(),
```

Comment:
With this new structure why do we have a maximum number of messages per block?

Reply:
I think we had a max before as well... Yup:

```
#[pallet::storage]
	#[pallet::getter(fn get_messages)]
	pub(super) type Messages<T: Config> = StorageDoubleMap<
		_,
		Twox64Concat,
		BlockNumberFor<T>,
		Twox64Concat,
		SchemaId,
		BoundedVec<Message<T::MessagesMaxPayloadSizeBytes>, T::MaxMessagesPerBlock>,
		ValueQuery,
	>;
```

Reply:
I noticed that but we changed the data structure. I am curious why we still need it with this new data structure.

Reply:
If my understanding is wrong about why the pov grew then could we have left messages as is and only updated schemas to reduce the pov size of reading them?

Reply:
We have to update all of them to bring down PoV

Reply:
About the `MaxMessagesPerBlock` check in that particular line we don't need it, only argument for adding it was to exactly get the same response of the old extrinsic. But i can be removed if everyone onboard

Reply:
I think if we can get rid of code that is not needed it's probably good code hygiene to remove it.